src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -0,0 +1,696 @@
+                    startIndex = endIndex + 1;
+                    endIndex = pathEnvVar.IndexOf(':', startIndex);
+                    if (endIndex < 0) // no ':' found
+                    {
+                        break;
+                    }
+                    if (endIndex <= startIndex + 1) // resulting string would be empty
+                    {
+                        continue;
+                    }
+
+                    // Extract it and see if our file lives there
+                    string subPath = pathEnvVar.Substring(startIndex, endIndex - startIndex);
+                    path = Path.Combine(subPath, filename);
+                    if (File.Exists(path))
+                    {
+                        return path;
+                    }
+                }
+            }
+
+            // Could not find the file
+            throw new Win32Exception(Interop.Errors.ENOENT);
+        }
+
+        /// <summary>Gets the path to the current executable, or null if it could not be retrieved.</summary>
+        private static string GetExePath()
+        {
+            // Determine the maximum size of a path
+            int maxPath = -1;
+            Interop.libc.GetPathConfValue(ref maxPath, Interop.libc.PathConfNames._PC_PATH_MAX, Interop.libc.DEFAULT_PC_PATH_MAX);
+
+            // Start small with a buffer allocation, and grow only up to the max path
+            for (int pathLen = 256; pathLen < maxPath; pathLen *= 2)
+            {
+                // Read from procfs the symbolic link to this process' executable
+                byte[] buffer = new byte[pathLen + 1]; // +1 for null termination
+                int resultLength = (int)Interop.libc.readlink(Interop.procfs.SelfExeFilePath, buffer, (IntPtr)pathLen);
+
+                // If we got one, null terminate it (readlink doesn't do this) and return the string
+                if (resultLength > 0)
+                {
+                    buffer[resultLength] = (byte)'\0';
+                    return Encoding.UTF8.GetString(buffer, 0, resultLength);
+                }
+
+                // If the buffer was too small, loop around again and try with a larger buffer.
+                // Otherwise, bail.
+                if (resultLength == 0 || Marshal.GetLastWin32Error() != Interop.Errors.ENAMETOOLONG)
+                {
+                    break;
+                }
+            }
+
+            // Could not get a path
+            return null;
+        }
+
+        /// <summary>Computes a time based on a number of ticks since boot.</summary>
+        /// <param name="ticksAfterBoot">The number of ticks since boot.</param>
+        /// <returns>The converted time.</returns>
+        internal static DateTime BootTimeToDateTime(ulong ticksAfterBoot)
+        {
+            // Read procfs to determine the system's uptime, aka how long ago it booted
+            string uptimeStr = File.ReadAllText(Interop.procfs.ProcUptimeFilePath);
```

Comment:
I'm sure the sniffing would do the right thing, but can you pass an explicit encoding here (UTF8, I assume?)